Breadcrumbs.py

```diff
@@ -108,4 +119,182 @@ def get_points_by_row(row):
               breadcrumbs[index_of_breadcrumb] = breadcrumbs[index_of_breadcrumb][0:length_of_trim]
           break
       number_of_characters_left -= current_length
-    view.set_status('breadcrumbs',separator.join(breadcrumbs+['']))
+
+  return breadcrumbs
+
+
+def copy(view, text):
+  sublime.set_clipboard(html.parser.HTMLParser().unescape(text))
+  view.hide_popup()
+  sublime.status_message('Breadcrumbs copied to clipboard')
+
+
+class BreadcrumbsCommand(sublime_plugin.EventListener):
+
+  def on_selection_modified(self, view):
+    settings = sublime.load_settings('Breadcrumbs.sublime-settings')
+    view.erase_status('breadcrumbs')
+    if settings.get('breadcrumbs_statusbar', True):
+      separator = settings.get('breadcrumbs_separator', u' › ')
+      current_row = view.rowcol(view.sel()[0].b)[0]
+      breadcrumbs = make_breadcrumbs(view, current_row, False)
+      if breadcrumbs is None or len(breadcrumbs) < 1:
+        view.erase_status('breadcrumbs')
+      else:
+        view.set_status('breadcrumbs', separator.join(make_breadcrumbs(view, current_row, True)))
+    else:
+      view.erase_status('breadcrumbs')
+
+
+class BreadcrumbsPopupCommand(sublime_plugin.TextCommand):
+
+  def run(self, edit):
+    stylesheet = '''
+      p {
+        margin-top: 0;
+      }
+      a {
+        font-family: system;
+        font-size: 1.05rem;
+      }
+    '''
+
+    template = '''
+      <body id=show-scope>
+          <style>{stylesheet}</style>
+          <p>{breadcrumbs}</p>
+          <a href="{breadcrumbs_string}">Copy</a>
+      </body>
+    '''
+
+    settings = sublime.load_settings('Breadcrumbs.sublime-settings')
+    separator = settings.get('breadcrumbs_separator', u' › ')
+    view = self.view
+    current_row = view.rowcol(view.sel()[0].b)[0]
+    breadcrumbs = make_breadcrumbs(view, current_row, False)
+    if len(breadcrumbs) > 0:
+      breadcrumbs_element = '<br>'.join(breadcrumbs + [''])
+    else:
+      breadcrumbs_element = '<em>None</em>'
+
+    body = template.format(
+        breadcrumbs=breadcrumbs_element,
+        breadcrumbs_string=html.escape(separator.join(breadcrumbs), quote=True),
+        stylesheet=stylesheet
+    )
+    view.show_popup(body, max_width=512, on_navigate=lambda x: copy(view, x))
```

Comment:
Ok, sounds good, I'll try that. I'm learning most of this as I go (which is why this PR has been so awesome already), so I have been able to optimise some areas while others are still just the way I copy-pasted them.